site/assets/js/application.js

```diff
@@ -101,6 +101,16 @@
     })
   }
 
+  var rootElementExample = document.getElementById('root-element-example')
+  if (rootElementExample) {
+    var rootElementModal = new bootstrap.Modal(rootElementExample, {
+      rootElement: document.getElementById('another-element')
+    })
+    document.getElementById('root-element-example-btn').addEventListener('click', function () {
+      rootElementModal.show()
+    })
+  }
+
```

Comment:
I think, you can omit this and put a `data-bs-parent` attribute  (instead of `rootElement`)